File: represent/bench/java_flat/FormalBench_0242__FormalBenchHF__diverse__rule_1__IsNonagonal_llm_1.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;

class IsNonagonal {

	public static int isNonagonal(int num) {
		int[] arr = new int[num + 1];
		arr[0] = 0;
		for (int p = 0; p <= num - 1; p++) {
			arr[p + 1] = arr[p] + 7 * p + 1;
		}
		return arr[num];
	}
}

File: represent/bench/c_flat/FormalBench_0242__FormalBenchHF__diverse__rule_1__IsNonagonal_llm_1.c
#include <stdlib.h>

int isNonagonal(int num);

int isNonagonal(int num) {

		int *arr = (int *)malloc(sizeof(int) * (num + 1));
int arr_len = num + 1;
		arr[0] = 0;
		for (int p = 0; p <= num - 1; p++) {
			arr[p + 1] = arr[p] + 7 * p + 1;
		}
		return arr[num];
}
